core/src/main/java/software/amazon/awssdk/http/pipeline/stages/ApplyUserAgentStage.java

```diff
@@ -40,11 +48,37 @@ public ApplyUserAgentStage(HttpClientDependencies dependencies) {
     public SdkHttpFullRequest.Builder execute(SdkHttpFullRequest.Builder request, RequestExecutionContext context)
             throws Exception {
         RequestClientOptions opts = context.requestConfig().getRequestClientOptions();
-        if (opts != null) {
-            return request.header(HEADER_USER_AGENT, RuntimeHttpUtils
-                    .getUserAgent(clientConfig, opts.getClientMarker(RequestClientOptions.Marker.USER_AGENT)));
-        } else {
-            return request.header(HEADER_USER_AGENT, RuntimeHttpUtils.getUserAgent(clientConfig, null));
+        String userAgent = opts != null
+                           ? getUserAgent(clientConfig, opts.getClientMarker(RequestClientOptions.Marker.USER_AGENT))
+                           : getUserAgent(clientConfig, null);
+
+        return request.header(HEADER_USER_AGENT, userAgent);
+    }
+
+    private String getUserAgent(ClientConfiguration config, final String userAgentMarker) {
```

Comment:
is userAgentMarker the requestLevel user agent? If so can we rename it?

Reply:
Not sure, I copied this code from the `SdkRuntimeUtils`. I'll look into it.